earth_enterprise/src/fusion/gecombineterrain/gecombineterrain.cpp

```diff
@@ -225,35 +230,80 @@ int main(int argc, char **argv) {
     options.opt("output", outdir);
     options.opt("indexversion", index_version);
     options.opt("sortbuf", sortbuf);
+    options.opt("numCompressThreads",numCompressThreads,
+               &khGetopt::RangeValidator<uint32, 1, kMaxNumJobsLimit_2>);
     options.opt("numcpus", numcpus,
                 &khGetopt::RangeValidator<uint32, 1, kMaxNumJobsLimit_2>);
     PERF_CONF_LOGGING( "proc_exec_config_cli_numcpus", taskName, numcpus );
     options.opt("read_cache_max_blocks", read_cache_max_blocks,
                 &khGetopt::RangeValidator<uint32, 0, 1024>);
     options.opt("read_cache_block_size", read_cache_block_size,
                 &khGetopt::RangeValidator<uint32, 1, 1024>);
-
     if (!options.processAll(argc, argv, argn)) {
       usage(progname);
     }
     if (help) {
       usage(progname);
     }
+
     if (argn == argc) {
-      usage(progname, "No input indexes specified");
+      usage(progname, "No input indices specified");
+    }
+    uint cmdDefaultCPUs = CommandlineNumCPUsDefault();
+    uint numavailable = GetMaxNumJobs();
+
+    /* 
+        Choosing number of CPUs:
+
+        numcpus: initially, the command line parameter --numcpus
+	
+        CommandLineNumCPUsDefault(): defined in /etc/opt/google/systemrc via maxjobs tag
+
+        GetMaxNumJobs(): the number of available jobs, defined as the minimum value of either:
+        1. the currently available jobs or the default number of jobs i.e. kMaxNumJobs 
+            (is 8) multiplied by KH_MAX_NUM_JOBS_COEFF, which is a compiler option
+        2. kMaxNumJobsDefault = 8
+        3. environment variable, KH_GOOGLE_MAX_NUM_JOBS, if defined
+        4. the number of currently available cpus, obtained through sysconf
+
+        The methodology first checks if the --numcpus flag was present, if
+        it was, it will go with what was passed, no matter the number. If
+        it was not, it will take the minimum value of what is defined in the
+        systemrc file and what is currently available (values described 
+        above). If the maxjobs tag in systemrc is invalid (i.e. less than
+        1) it will default to the available CPUs.
+
+         
+        --numcpus is deprecated in favor of --numCompressThreads. One
+        or the other may be passed in, but not both.
+    */
+ 
+    if (numCompressThread)
```

Comment:
This is a compile error - should be `numCompressThreads` (with an 's' on the end).

Reply:
I can fix this quickly online if you like.